res/kiwixAPI.js

```diff
@@ -0,0 +1,43 @@
+import "core-js/stable";
+import "regenerator-runtime/runtime";
+
+let zim = {toc: {}};
+
+function getSectionList(){
+    let sectionList = [];
+    for (i = 0; i < document.querySelectorAll('h1, h2, h3, h4, h5, h6').length; i++) {
+        let headerObject = document.querySelectorAll('h1, h2, h3, h4, h5, h6')[i];
+        if (headerObject.id === "") {
+            headerObject.id = "documentparserid" + i;
+        }
+        sectionList.push(headerObject);
+    }
+    return sectionList;
+}
+
+zim.toc.hasTableOfContent = function() {
+    return getSectionList().length > 0 ? true : false;
```

Comment:
This call (complex) `getSectionList` just to know if a `h*` tag is present.
You can directly test the value of `document.querySelectorAll('h1, h2, h3, h4, h5, h6').length`. Or even better test that on header tag is present without getting all header tags with specific js api I don't know.